### PR TITLE
Windows 2.x GitHub Action

### PR DESCRIPTION
**Pull request by scrawfor99**
Basic setup for creating a GitHub Action to run test on push, merge, etc. for Windows.

This is a draft PR and there are several things about it that require further work. Namely: 
- It is not clear how the installation files are fetched when Docker is not used (without Docker being used, cannot call Docker hub)
- The plugin installation process is not successful and the current version makes use of the broken Windows core version where security does not function. 
- The final integration tests are built with Docker in mind so this requires migration. 

### Testing

So far the following functionality has been manually checked on a Windows instance:
1. Enable longer filenames
2. Gradle assemble security 
3. Hardcoded 2.3.0 Snapshot pull (command prompt)
4. Zip extraction and old zip deletion --> The extracted file organization is that of a snapshot so is built for Docker
5. Starting .\bin\opensearch.bat (powershell) --> This currently fails due to bad security plugin install 
6. Curl get https://localhost:9200 -u 'admin:admin' -k -v (command prompt) 
The intermediate steps have not been confirmed to work yet.
### Check List
- [ ] ~New functionality includes testing~
- [X] New functionality has been documented
- [X] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: .github/workflows/plugin_install.yml [hunk 1]
```diff
@@ -3,63 +3,107 @@ name: Plugin Install
...
+        uses: actions/checkout@v2
+
+      - name: Set up JDK
+        uses: actions/setup-java@v1
+        with:
+          java-version: 11
+
+      - name: Download OpenSearch Core
+        run: |
+          curl https://artifacts.opensearch.org/snapshots/core/opensearch/2.4.0-SNAPSHOT/opensearch-min-2.4.0-SNAPSHOT-windows-x64-latest.zip
```
Comment (peternied) [inline .github/workflows/plugin_install.yml:87]: Checkout this stackoverflow question about downloading files on windows https://stackoverflow.com/a/41862814/533057

Comment (peternied) [reply .github/workflows/plugin_install.yml:87]: Note; this will need to user powershell to invoke

**Comment by peternied** [comment]
@scrawfor99 I know you are working on this - please @mention me when you'd like me to take another look

**Comment by scrawfor99** [comment]
> @scrawfor99 I know you are working on this - please @mention me when you'd like me to take another look

@peternied if you want to lend a second set of eyes that would be appreciated. I am running into a path issue that is causing me some problems because I am not sure how to view the paths that GitHub makes so I do not know what to specify the install locations etc. as

**Review by DarshitChanpura — COMMENTED** [review]
Looks good overall @scrawfor99 !
I do see a lot of commits with same message, ideally each commit message should be unique and convey a purpose for that commit.

**Comment by scrawfor99** [comment]
> Looks good overall @scrawfor99 ! I do see a lot of commits with same message, ideally each commit message should be unique and convey a purpose for that commit.

The commits have the same message because I cannot test the github action without running it so I am having to rerun it many times since the steps are not one:one for what works on the server versus what works for Github.

File: .github/workflows/plugin_install.yml [hunk 2]
```diff
@@ -3,63 +3,122 @@ name: Plugin Install
...
+          Invoke-WebRequest https://artifacts.opensearch.org/snapshots/core/opensearch/2.4.0-SNAPSHOT/opensearch-min-2.4.0-SNAPSHOT-windows-x64-latest.zip -Outfile opensearch-2.4.0.zip
+          tar -xzf opensearch-2.4.0.zip 
+          del opensearch-2.4.0.zip
+          Rename-Item opensearch-* Opensearch
+         
+           
+
+      - name: Build Security Plugin with Gradle
+        run: |
+          .\gradlew.bat assemble
```
[the remark this answers is not in the document]
Comment (scrawfor99) [reply .github/workflows/plugin_install.yml:95]: What does this argument do?

**Comment by scrawfor99** [comment]
I think something is wrong with the SSL certificates that are created for the testing. The only way to serve a request manually is by enforcing all certificates to be ignored, and that is not an option on GitHub Actions.

File: .github/workflows/plugin_install.yml [hunk 3]
```diff
@@ -3,63 +3,122 @@ name: Plugin Install
...
+        env:
+          _JAVA_OPTIONS: -Xmx4096M
```
[the remark this answers is not in the document]
Comment (scrawfor99) [reply .github/workflows/plugin_install.yml:97]: I ran it on a 32bit platform remotely and it seemed okay but I will look for any issues that come up!

**Pull request closed without merging**